openssl/src/x509/mod.rs

```diff
@@ -2054,6 +2055,30 @@ impl GeneralName {
             Ok(GeneralName::from_ptr(gn))
         }
     }
+
+    pub(crate) fn new_other_name(
+        oid: Asn1Object,
+        value: *mut ASN1_TYPE,
+    ) -> Result<GeneralName, ErrorStack> {
+        unsafe {
+            ffi::init();
+            let gn = cvt_p(ffi::GENERAL_NAME_new())?;
+            (*gn).type_ = ffi::GEN_OTHERNAME;
+
+            if let Err(e) = cvt(ffi::GENERAL_NAME_set0_othername(
+                gn,
+                oid.as_ptr().cast(),
+                value,
+            )) {
```

Comment:
Unfortunately this isn't sound -- because `build` takes `&self` you can call it multiple times, in which case the `value` ptr will be reused.

Reply:
handled as already commented somewhere else